Bug fix: Convert dateTimes

saulius-saltenis-Devoteam opened the pull request:
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context.

I store UTC timestamps in DB, and pass ISO timestamps for filtering operations.
When ISO timestamp is passed, it's first converted to `DateTime` with additional local timezone offset. Later, it's `Kind` is set to UTC. This only changes the `Kind` property, retaining the previously added timezone ticks.
![image](https://github.com/user-attachments/assets/7c8673b4-3dca-475e-b358-a524d80385fe)
![image](https://github.com/user-attachments/assets/e393bf69-81ef-4c36-98d9-b15e12e76c71)
![image](https://github.com/user-attachments/assets/2634868d-a22a-48de-87f5-18a3b680ac8c)

For this reason, the following request:
![image](https://github.com/user-attachments/assets/9bf448da-7732-4577-a4fb-d4962a3f94a8)
Turns into this query:
![image](https://github.com/user-attachments/assets/ae64a41d-292a-4e6c-8d5f-32d846f9a5ca)


This behavior can be easily reproduced with the following sample, when environment has a timezone:
```csharp
var dateTime = DateTime.Now;
Console.WriteLine(dateTime);

var utcDateTime = DateTime.SpecifyKind(dateTime, DateTimeKind.Utc);
Console.WriteLine(utcDateTime);

var utcDateTime2 = dateTime.ToUniversalTime();
Console.WriteLine(utcDateTime2);
```

![image](https://github.com/user-attachments/assets/12442fcc-e451-4e4f-be7d-89aed387a19c)


Fixes # (issue)
-

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have made corresponding changes to the documentation
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] New and existing unit tests pass locally with my changes

alirezanet reviewed (commented):
Hi @saulius-saltenis-Devoteam,
Looks good to me, but it would be nice if you add a couple of unit-tests to validate the old and new behavior

saulius-saltenis-Devoteam commented:
Will add either later today, or tomorrow

saulius-saltenis-Devoteam commented:
Added a test for ensuring that the final query filters with same timestamp as passed via input.
Added two more tests below, that showcase the problem with past conversions.

alirezanet reviewed (approved):
LGTM 💐👌
